src/basics.md

```diff
@@ -1770,6 +1771,69 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-read-environment-variable]: #ex-read-environment-variable
+<a name="ex-read-environment-variable"></a>
+## Read environment variable
+[![std-badge]][std] [![cat-filesystem-badge]][cat-filesystem]
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+
+use std::env;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::io::{BufReader, BufRead};
+
+# error_chain! {
+#     foreign_links {
+#         Utf8(std::str::Utf8Error);
+#         AddrParse(std::net::AddrParseError);
+#     }
+# }
+
+fn create_sample_config(path: &PathBuf) -> std::io::Result<()> {
+    let mut output = File::create(path)?;
+    write!(output, "Sample content")?;
+
+    Ok(())
+}
+
+fn set_env_variable() {
+  env::set_var("CONFIG_FILE", "/tmp/config.toml");
```

Comment:
This is not a cross platform solution.  I'm looking into the way the cookbook wants to recommend implementing file system abstractions.  If you have any ideas, I'm interested.  Right now I'm reading https://github.com/rust-lang-nursery/cli-wg/issues/10